src/stores/api/wpcom-api.ts

```diff
@@ -0,0 +1,52 @@
+import { createApi } from '@reduxjs/toolkit/query/react';
+import type { FetchBaseQueryError } from '@reduxjs/toolkit/query';
+import type WPCOM from 'wpcom';
+
+export interface WelcomeMessageResponse {
+	messages: string[];
+	example_prompts: string[];
+}
+
+let wpcomClient: WPCOM | undefined;
+
+export const setWpcomClient = ( client: WPCOM | undefined ) => {
+	wpcomClient = client;
+};
+
+const wpComBaseQuery = async ( args: { path: string; apiNamespace?: string } ) => {
+	if ( ! wpcomClient?.req ) {
+		return {
+			error: {
+				status: 401,
+				data: 'Not authenticated',
+			} as FetchBaseQueryError,
+		};
+	}
+
+	try {
+		const response = await wpcomClient.req.get( args );
+		return { data: response };
+	} catch ( error ) {
+		return {
+			error: {
+				status: 500,
+				data: error,
+			} as FetchBaseQueryError,
+		};
+	}
+};
```

Comment:
```suggestion
const wpcomBaseQuery: BaseQueryFn<
	{ path: string; apiNamespace?: string },
	unknown,
	FetchBaseQueryError
> = async ( args ) => {
	if ( ! wpcomClient?.req ) {
		const error: FetchBaseQueryError = {
			status: 401,
			data: 'Not authenticated',
		};

		return { error };
	}

	try {
		const response = await wpcomClient.req.get( args );
		return { data: response };
	} catch ( error ) {
		const fetchQueryError: FetchBaseQueryError = {
			status: 500,
			data: error,
		};

		return { error: fetchQueryError };
	}
};
```
A couple of thoughts about the base query function:
1. Let's use type definitions for the error objects instead of `as`
2. Let's use a type definition for the entire function instead of for just the args
3. Subjective nit is also that I think `wpcomClient` reads better than `wpComClient`